### PR TITLE
feat: Add the git-log-prs script

### PR DESCRIPTION
**Pull request by mkmik**
__Scenarios__

You may know the git sha of what's running in the staging cluster,
and you may know the sha of the latest built image is, but will upgrading to a new
version include the PRs I'm interested in?

Or alternatively: I noticed in the dashboard that IOx was working fine until
we rolled out a new version. Which PRs were included in this new rollout?

__Description__

Getting the answers to the scenarios above is surprisingly hard, because scanning
our git history is complicated by the fact that just about anybody uses a different
merging technique.

This script does the right magic to skim through all that cruft and get the answers.

__Demo__

```console
$ ./scripts/git-log-prs 8376983b74311df970339e106c62ce4038b20e5f..
1330
1336
$ ./scripts/git-log-prs 8376983b74311df970339e106c62ce4038b20e5f.. --titles
1330 feat: Make background task period configurable
1336 feat: Build a perf_image image for every commit in main
```

### CODE REVIEW

File: scripts/git-log-prs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    usage
+  fi
+
+  local rev_range="${positionals[0]}"
+
+  git log \
+      --committer='GitHub <noreply@github.com>' \
+      --pretty=format:'%s' \
+      "${rev_range}" \
+    | sed 's/Merge pull request #\([0-9]*\).*/\1/' \
```
Comment (alamb) [inline scripts/git-log-prs:63]: This requires that we have used `kodiakHQ` right as it is looking for the merge message that Kodiak creates?

**Review by alamb — APPROVED** [review]
Unix wizardry FTW

> our git history is complicated by the fact that just about anybody uses a different
merging technique.

Yes this is why I like using `squash-merge` for prs in github so there is a 1-1 correspondence between PR and commit on master

During a PR review, having the work broken into multiple commits can definitely help the review process and is valuable. 

After the PR is merged, I can not think of any time I have ever wanted the individual commit history on main. I *can* think of many times I have wanted to know "what PR changed this line" which our current merge method makes challenging to determine

I am fairly sure that @e-dard  and @domodwyer  feel strongly about keeping individual commit history on master otherwise I would recommend going to the "1 master commit per PR" model

File: scripts/git-log-prs (hunk 1 above)
Comment (mkmik) [reply scripts/git-log-prs:63]: yes, there *is* a solution for that but I'm too lazy to do it now

Comment (mkmik) [reply scripts/git-log-prs:63]: no sorry, this requires that GitHub merge button performs the merge.

It doesn't matter if kodiak or a human clicks on it.

It won't work if and admin manually merges and pushes to `main`

Comment (mkmik) [reply scripts/git-log-prs:63]: (because github is the one issue the merge commit, or the one performing the squash)

Comment (mkmik) [reply scripts/git-log-prs:63]: I'm not 100% sure what happens when the "rebase" mode is chosen and it results in a perfect fast-forward merge

Comment (mkmik) [reply scripts/git-log-prs:63]: will investigate and improve as we go

**Pull request merged**